front/lib/swr/data_source_views.ts

```diff
@@ -211,3 +212,28 @@ export function useDataSourceViewConnectorConfiguration({
     isDocumentError: error,
   };
 }
+
+export function useDataSourceViewTables({
+  dataSourceView,
+  workspaceId,
+}: {
+  dataSourceView: DataSourceViewType | null;
+  workspaceId: string;
+}) {
+  const tablesFetcher: Fetcher<ListTablesResponseBody> = fetcher;
+  const disabled = !dataSourceView;
+
+  const { data, error, mutate } = useSWRWithDefaults(
+    disabled
+      ? null
```

Comment:
@Fraggle does this follow the new approach?

Reply:
If `dataSourceView` has to be nullable, yes. Passing `{disabled}` as an option wouldn't change anything (mutate would not work etc..)